micropip/logging.py

```diff
@@ -88,12 +88,13 @@ def _set_formatter_once() -> None:
 
 
 class LoggerWrapper:
-    __slots__ = ("logger", "_orig_level")
 
-    logger: logging.Logger
+    # need a default value because of __getattr__/__setattr__
+    logger: logging.Logger = None  # type: ignore[assignment]
 
     def __init__(self, logger: logging.Logger):
-        self.logger = logger
+        # Bypassing __setattr__ by setting attributes directly in __dict__
+        self.__dict__["logger"] = logger
```

Comment:
What's the point of this? I don't understand.

Reply:
`self.logger = ...` call setattr, which itself, need  `self.logger`, (in `getattr(self.logger, ...,...)`, though `self.logger` is not defined so goes through getattr (infinite recursion). OTOH `self.__dict__` exists, so all fine. 

Basically #138 should be merged first, it fixes the failure on main, and then I'll rebased my own PRs on top of #138.